packages/datetime/test/dateInputTests.tsx

```diff
@@ -403,6 +403,50 @@ describe("<DateInput>", () => {
             assertDateEquals(onError.args[0][0], new Date(value));
         });
 
+        it("typing in a date out of range and pressing enter displays the error message", () => {
+            const rangeMessage = "RANGE ERROR";
+            const onError = sinon.spy();
+            const wrapper = mount(
+                <DateInput
+                    {...DATE_FORMAT}
+                    defaultValue={new Date(2015, Months.MAY, 1)}
+                    minDate={new Date(2015, Months.MARCH, 1)}
+                    onError={onError}
+                    outOfRangeMessage={rangeMessage}
+                />,
+            );
+            const outOfRangeDate = "2/1/2030";
+            wrapper
+                .find("input")
+                .simulate("change", { target: { outOfRangeDate } })
+                .simulate("keydown", { which: Keys.ENTER });
+
+            assert.strictEqual(wrapper.find(InputGroup).prop("intent"), Intent.DANGER);
+            assert.strictEqual(wrapper.find(InputGroup).prop("value"), rangeMessage);
+        });
+
+        it("typing in a date out of range and pressing enter calls onError with invalid date", () => {
```

Comment:
please merge these tests since they only differ in the final assertions. no need to duplicate all the setup. 

"... displays error message and calls onError"

Reply:
https://github.com/palantir/blueprint/projects/2#card-1305041